### ISSUE TITLE
Composum not compatbile with Java 11: Unknown protocol: jrt

### ISSUE THREAD

**Issue by stefanseifert** [ist-dresden/composum#170]
Steps to reproduce:
- download Sling 11: http://ftp.halifax.rwth-aachen.de/apache/sling/org.apache.sling.starter-11.jar
- Start **using Java 11**: `java -jar org.apache.sling.starter-11.jar`
- Login as admin/admin
- Try to access composum: http://localhost:8080/bin/browser.html

expected: composum console should open
actual: an error occurs

```
28.05.2019 17:33:55.204 *ERROR* [127.0.0.1 [1559057635203] GET /bin/browser.html HTTP/1.1] org.apache.sling.engine.impl.SlingRequestProcessorImpl service: Uncaught SlingException
java.lang.IllegalStateException: Unknown protocol: jrt
	at org.apache.felix.framework.URLHandlersStreamHandlerProxy.toExternalForm(URLHandlersStreamHandlerProxy.java:482)
	at org.apache.felix.framework.URLHandlersStreamHandlerProxy.toExternalForm(URLHandlersStreamHandlerProxy.java:474)
	at java.base/java.net.URL.toExternalForm(URL.java:1001)
	at java.base/java.net.URL.toString(URL.java:987)
	at org.apache.felix.framework.URLHandlersStreamHandlerProxy.openConnection(URLHandlersStreamHandlerProxy.java:267)
	at java.base/java.net.URL.openConnection(URL.java:1051)
	at java.base/java.net.URLClassLoader.getResourceAsStream(URLClassLoader.java:327)
	at org.apache.sling.commons.compiler.impl.EclipseJavaCompiler$CompileContext.isPackage(EclipseJavaCompiler.java:385) [org.apache.sling.commons.compiler:2.3.6]
	at org.apache.sling.commons.compiler.impl.EclipseJavaCompiler$CompileContext.isPackage(EclipseJavaCompiler.java:360) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.lookup.ModuleBinding.getVisiblePackage(ModuleBinding.java:543) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.lookup.ModuleBinding.getPackage(ModuleBinding.java:606) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.lookup.PackageBinding.findPackage(PackageBinding.java:138) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.lookup.PackageBinding.getTypeOrPackage(PackageBinding.java:275) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.lookup.Scope.getPackage(Scope.java:2929) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.ast.QualifiedTypeReference.getTypeBinding(QualifiedTypeReference.java:104) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.ast.TypeReference.internalResolveType(TypeReference.java:495) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.ast.TypeReference.resolveType(TypeReference.java:586) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.ast.TypeReference.resolveType(TypeReference.java:582) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.lookup.SourceTypeBinding.resolveTypeFor(SourceTypeBinding.java:1766) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.lookup.SourceTypeBinding.fields(SourceTypeBinding.java:933) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.lookup.SourceTypeBinding.internalFaultInTypeForFieldsAndMethods(SourceTypeBinding.java:902) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.lookup.SourceTypeBinding.faultInTypesForFieldsAndMethods(SourceTypeBinding.java:898) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.lookup.CompilationUnitScope.faultInTypes(CompilationUnitScope.java:509) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.Compiler.process(Compiler.java:878) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.Compiler.processCompiledUnits(Compiler.java:568) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:468) [org.apache.sling.commons.compiler:2.3.6]
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:419) [org.apache.sling.commons.compiler:2.3.6]
	at org.apache.sling.commons.compiler.impl.EclipseJavaCompiler.compile(EclipseJavaCompiler.java:234) [org.apache.sling.commons.compiler:2.3.6]
	at org.apache.sling.scripting.jsp.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:115) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.jsp.jasper.compiler.Compiler.compile(Compiler.java:304) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.jsp.jasper.compiler.Compiler.compile(Compiler.java:282) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.jsp.jasper.compiler.Compiler.compile(Compiler.java:269) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.jsp.jasper.JspCompilationContext.compile(JspCompilationContext.java:501) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.jsp.jasper.servlet.JspServletWrapper.prepareServlet(JspServletWrapper.java:427) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.jsp.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:486) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.jsp.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:449) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.jsp.JspScriptEngineFactory.callJsp(JspScriptEngineFactory.java:342) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.jsp.JspScriptEngineFactory.access$100(JspScriptEngineFactory.java:97) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.jsp.JspScriptEngineFactory$JspScriptEngine.eval(JspScriptEngineFactory.java:603) [org.apache.sling.scripting.jsp:2.3.4]
	at org.apache.sling.scripting.core.impl.DefaultSlingScript.call(DefaultSlingScript.java:388) [org.apache.sling.scripting.core:2.0.54]
	at org.apache.sling.scripting.core.impl.DefaultSlingScript.eval(DefaultSlingScript.java:184) [org.apache.sling.scripting.core:2.0.54]
	at org.apache.sling.scripting.core.impl.DefaultSlingScript.service(DefaultSlingScript.java:491) [org.apache.sling.scripting.core:2.0.54]
	at org.apache.sling.engine.impl.request.RequestData.service(RequestData.java:552) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.filter.SlingComponentFilterChain.render(SlingComponentFilterChain.java:44) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:82) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.SlingRequestProcessorImpl.processComponent(SlingRequestProcessorImpl.java:282) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.SlingRequestProcessorImpl.dispatchRequest(SlingRequestProcessorImpl.java:322) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.request.SlingRequestDispatcher.dispatch(SlingRequestDispatcher.java:211) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.request.SlingRequestDispatcher.forward(SlingRequestDispatcher.java:146) [org.apache.sling.engine:2.6.14]
	at com.composum.sling.core.servlet.AbstractConsoleServlet.doGet(AbstractConsoleServlet.java:76) [com.composum.core.commons:1.9.2]
	at org.apache.sling.api.servlets.SlingSafeMethodsServlet.mayService(SlingSafeMethodsServlet.java:266) [org.apache.sling.api:2.18.4]
	at org.apache.sling.api.servlets.SlingSafeMethodsServlet.service(SlingSafeMethodsServlet.java:342) [org.apache.sling.api:2.18.4]
	at org.apache.sling.api.servlets.SlingSafeMethodsServlet.service(SlingSafeMethodsServlet.java:374) [org.apache.sling.api:2.18.4]
	at org.apache.sling.engine.impl.request.RequestData.service(RequestData.java:552) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.filter.SlingComponentFilterChain.render(SlingComponentFilterChain.java:44) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:82) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.SlingRequestProcessorImpl.processComponent(SlingRequestProcessorImpl.java:282) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.filter.RequestSlingFilterChain.render(RequestSlingFilterChain.java:49) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:82) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.debug.RequestProgressTrackerLogFilter.doFilter(RequestProgressTrackerLogFilter.java:107) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:72) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.i18n.impl.I18NFilter.doFilter(I18NFilter.java:131) [org.apache.sling.i18n:2.5.14]
	at org.apache.sling.engine.impl.filter.AbstractSlingFilterChain.doFilter(AbstractSlingFilterChain.java:72) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.SlingRequestProcessorImpl.doProcessRequest(SlingRequestProcessorImpl.java:151) [org.apache.sling.engine:2.6.14]
	at org.apache.sling.engine.impl.SlingMainServlet.service(SlingMainServlet.java:234) [org.apache.sling.engine:2.6.14]
	at org.apache.felix.http.base.internal.handler.ServletHandler.handle(ServletHandler.java:120) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.base.internal.dispatch.InvocationChain.doFilter(InvocationChain.java:86) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.sling.i18n.impl.I18NFilter.doFilter(I18NFilter.java:131) [org.apache.sling.i18n:2.5.14]
	at org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:135) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.base.internal.dispatch.InvocationChain.doFilter(InvocationChain.java:81) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.sling.engine.impl.log.RequestLoggerFilter.doFilter(RequestLoggerFilter.java:72) [org.apache.sling.engine:2.6.14]
	at org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:135) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.base.internal.dispatch.InvocationChain.doFilter(InvocationChain.java:81) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.sling.engine.impl.parameters.RequestParameterSupportConfigurer.doFilter(RequestParameterSupportConfigurer.java:63) [org.apache.sling.engine:2.6.14]
	at org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:135) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.base.internal.dispatch.InvocationChain.doFilter(InvocationChain.java:81) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.base.internal.dispatch.Dispatcher$1.doFilter(Dispatcher.java:146) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.base.internal.whiteboard.WhiteboardManager$2.doFilter(WhiteboardManager.java:1014) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.sslfilter.internal.SslFilter.doFilter(SslFilter.java:97) [org.apache.felix.http.sslfilter:1.2.6]
	at org.apache.felix.http.base.internal.handler.PreprocessorHandler.handle(PreprocessorHandler.java:133) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.base.internal.whiteboard.WhiteboardManager$2.doFilter(WhiteboardManager.java:1020) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.base.internal.whiteboard.WhiteboardManager.invokePreprocessors(WhiteboardManager.java:1024) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.base.internal.dispatch.Dispatcher.dispatch(Dispatcher.java:91) [org.apache.felix.http.jetty:4.0.6]
	at org.apache.felix.http.base.internal.dispatch.DispatcherServlet.service(DispatcherServlet.java:49) [org.apache.felix.http.jetty:4.0.6]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:725) [org.apache.felix.http.servlet-api:1.1.2]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:535) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.Server.handle(Server.java:503) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765) [org.apache.felix.http.jetty:4.0.6]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683) [org.apache.felix.http.jetty:4.0.6]
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Sling 11 contains Composum 1.9.2, but same problem occurs when updating to Composum 1.9.4.

**Comment by ist-rw** [comment]
Seems to be the known bug of JDK 11.0.2 (https://bugs.openjdk.java.net/browse/JDK-8217364.). I'm using JDK 11.0.1 and it works fine.

**Comment by ist-rw** [comment]
The Java issue is still present in JDK 11.0.3 (Fix version: 11.0.5) but a downgrade to JDK 11.0.1 should help (https://www.oracle.com/technetwork/java/javase/downloads/java-archive-javase11-5116896.html).

**Comment by ist-rw** [comment]
A Felix upgrade to 6.0.2 with a workaround to fix this issue is also an option (https://issues.apache.org/jira/browse/FELIX-6035).

**Comment by stoerr** [comment]
I think the main issue is the Felix Bug https://issues.apache.org/jira/browse/FELIX-6035 ; it works on JDK 11.0.1 anyway because of the bug https://bugs.openjdk.java.net/browse/JDK-8217364 , which is fixed in later versions (11.0.3 and 12.0.1). Besides using JDK 11.0.1 you have also a couple of other options at the moment:

- you can add option --add-opens=java.base/sun.net.www.protocol.jrt=ALL-UNNAMED to the JVM options. I tried that for 11.0.3 and 12.0.1, though it does not look like something you should do on a production system. That's a hack suggested at https://issues.apache.org/jira/browse/NETBEANS-2160 .

- if you feel adventurous, you can also use https://github.com/ist-dresden/sling-org-apache-sling-starter/tree/jdk11minimalfix-felix6.0.2 where I just changed the org.apache.sling.launchpad.base version 6.0.1-2.6.32 to 6.0.2-2.6.36 , or build your own launchpad from the master branch of https://github.com/apache/sling-org-apache-sling-starter .
